aptos-move/framework/aptos-framework/sources/account/account.move

```diff
@@ -424,6 +443,185 @@ module aptos_framework::account {
         rotate_authentication_key_internal(account, new_auth_key);
     }
 
+    /// Adds an ED25519 backup key to a keyless account by converting the account's authentication key to a multi-key.
+    /// This function takes a keyless account (identified by having a keyless public key scheme) and adds an ED25519 backup key,
+    /// creating a multi-key that requires 1 signature from either key to authenticate.
+    ///
+    /// # Arguments
+    /// * `account` - The signer representing the keyless account
+    /// * `keyless_public_key` - The current keyless public key of the account
+    /// * `backup_key` - The ED25519 public key to add as a backup
+    /// * `backup_key_proof` - A signature from the backup key proving ownership
+    ///
+    /// # Aborts
+    /// * If the current public key is not a keyless public key
+    /// * If the provided keyless public key does not match the account's current authentication key
+    /// * If the keyless public key is not the original public key of the account
+    /// * If the backup key proof signature is invalid
+    ///
+    /// # Events
+    /// * Emits a `KeyRotationToMultiPublicKey` event with the new multi-key configuration
+    entry fun add_ed25519_backup_key_on_keyless_account(account: &signer, keyless_public_key: vector<u8>, backup_key: vector<u8>, backup_key_proof: vector<u8>) acquires Account {
+        // Check that the public key is a keyless public key by checking the scheme
+        // encoded in the first byte of the single key public key
+        let public_key_type = keyless_public_key[0];
+        assert!(public_key_type == 3 || public_key_type == 4, error::invalid_argument(ENOT_A_KEYLESS_PUBLIC_KEY));
+        let addr = signer::address_of(account);
+        let account_resource = &mut Account[addr];
+
+        // Check that the provided public key is the current public key of the account by comparing
+        // its authentication key to the account's authentication key
+        let auth_key = get_authentication_key_from_scheme_and_public_key_bytes(SINGLE_KEY_SCHEME, keyless_public_key);
+        assert!(
+            auth_key == account_resource.authentication_key,
+            error::invalid_argument(EWRONG_CURRENT_PUBLIC_KEY)
+        );
+
+        // Check that the keyless public key is the original public key of the account by comparing
```

Comment:
Is this check needed after the auth key check?

Reply:
Yes. We want to ensure the original keyless public key is part of the key.

We can unrestrict it later if desired.

Reply:
I don't get it. The auth key check earlier already verifies that the public key's hash matches the authkey. What extra value does this check with address add?

Reply:
It restricts the domain of the function to keyless accounts that still have the original key as a signer.

This allows us to emit the event indicating verification of the original key and the backup key